lib/cn/spill.c

```diff
@@ -244,6 +244,9 @@ cn_subspill(
             return 0; /* Nothing to do */
     }
 
+    if (!sctx->more || (key_obj_cmp(&sctx->curr->kobj, &ekobj) > 0 && !sctx->pt_set))
+        return 0; /* Nothing to do */
+
```

Comment:
You should hoist out the (key_obj_cmp() > 0) and save the result in a bool, then use the var in the expressions.
The optimizer might do that at some levels, but for sure this should help debug build run faster.  Plus I think it would be easier to read...